sycl/doc/extensions/experimental/sycl_ext_oneapi_in_order_queue_events.asciidoc

```diff
@@ -0,0 +1,148 @@
+= sycl_ext_oneapi_in_order_queue_events
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2024-2024 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 8 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+
+== Status
+
+This is an experimental extension specification, intended to provide early
+access to features and gather community feedback.  Interfaces defined in this
+specification are implemented in {dpcpp}, but they are not finalized and may
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+
+== Overview
+
+SYCL 2020 in-order queues allow for simple control of submission ordering, i.e.
+commands are executed in the order they are submitted. This extension adds two
+additional APIs for controlling in-order queues: Getting the event from the last
+command submission into the queue and setting an external event as an implicit
+dependence on the next command submitted to the queue.
```

Comment:
Reading this made me realize that we should try to identify the extension's target audience in the Overview section.  I created #12339 to clarify this in the specification template.  I think it would be appropriate to add a short paragraph like this to the Overview section:

> This extension exists to solve a specific problem, and a general solution is still being evaluated.  It is not recommended for general usage.

Reply:
Good call. The note has been added.